### PR TITLE
⬆ update lancet

### PR DESCRIPTION
**Pull request by migbro**
## Description

Part of a series of PRs to update each caller sub wf and workflow with the goal of bringing the overall super workflow to date with the latest annotation. One thing that might need finessing/discussion is the exome+ vs WXS vs straight WGS. Currently Ops seems to run the CDS file even if it's WXS. I am sure that is ignored, and hopefully can be run without when not needed?

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] [cavatica test run](https://cavatica.sbgenomics.com/u/kfdrc-harmonization/kf-reference-pipeline/tasks/964734b1-affa-4b9e-ba39-c241beb81175/)
- [ ] Test B

**Test Configuration**:
* Environment: cavatica
* Test files: see task

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

**Review by sickler-alex — APPROVED** [review]

**Review by dmiller15 — APPROVED** [review]

File: docs/kfdrc_lancet_standalone_wf.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+1. Determine run-time parameters based on whether data is WGS or exome/targeted
+1. Create any missing index files
+1. Convert input alignment file to bam and recalculate `MD` tags, if applicable. Recalculating `MD` tags allows for the lancet algorithm to leverage this tag for speed
+1. Generate interval lists for scatter jobs
+1. If supplied, generate a specialized interval list using a supplied bed file, plus regions from hits in Strelka2 and Mutect2 output
+1. Run the [lancet subworkflow](../sub_workflows/kfdrc_lancet_sub_wf.cwl) as described [here](../docs/kfdrc_lancet_sub_wf.md)
+1. Collect the outputs from the subworkflow
+
+## Extended Use Case
+Lancet uses a graph model to refine INDEL calls, which can be a very time-consuming process. Therefore, lancet is run in what I'd call an "Exome+" mode, based on the NYGC methods described [here](https://www.biorxiv.org/content/biorxiv/early/2019/04/30/623702.full.pdf). In short, regions from GENCODE gtf with feature annotations `exon`, `UTR`, and start/stop `codon` are used as intervals, as well as regions flanking hits from `strelka2` and `mutect2`. This preferred mode of processing WGS input
```
Comment (NHJohnson) [inline docs/kfdrc_lancet_standalone_wf.md:15]: Fragment at end?

File: docs/kfdrc_lancet_sub_wf.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ - `reference_dict`: `Homo_sapiens_assembly38.dict`
+ - `bed_invtl_split`: an array of bed intervals, would be created by a wrapper workflow
+### Source-specific inputs
+ - `window`
+   - `600`: exome+ mode (see [wrapper workflow for explanation]() )
+   - `500`: If run as pure WGS. Not recommended for performance reasons, see [wrapper workflow for explanation]()
+ - `padding`
+   - `300` or half window size used if exome+
+   - `25` if pure WGS
+### Worflow performance tuning
```
Comment (NHJohnson) [inline docs/kfdrc_lancet_sub_wf.md:65]: Workflow

File: docs/kfdrc_lancet_sub_wf.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+Occassionally, default memory values may not suffice to process the inputs.
```
Comment (NHJohnson) [inline docs/kfdrc_lancet_sub_wf.md:66]: Occasionally

File: sub_workflows/kfdrc_lancet_sub_wf.cwl [hunk 4]
```diff
@@ -17,16 +18,30 @@ inputs:
...
+  tool_name: {type: string?, doc: "String to describe what tool was run as part of file name", default: "lancet_somatic"}
+  # annotation vars
+  genomic_hotspots: { type: 'File[]?', doc: "Tab-delimited BED formatted file(s) containing hg38 genomic positions corresponding to hotspots" }
+  protein_snv_hotspots: { type: 'File[]?', doc: "Column-name-containing, tab-delimited file(s) containing protein names and amino acid positions corresponding to hotspots" }
+  protein_indel_hotspots: { type: 'File[]?', doc: "Column-name-containing, tab-delimited file(s) containing protein names and amino acid position ranges corresponding to hotspots" }
+  retain_info: {type: string?, doc: "csv string with INFO fields that you want to keep", default: "MS,FETS,HotSpotAllele"}
+  retain_fmt: {type: string?, doc: "csv string with FORMAT fields that you want to keep"}
+  maf_center: {type: string?, doc: "Sequencing center of variant called", default: "."}
+  add_common_fields: {type: boolean?, doc: "Set to true if input is a strelka2 vcf that hasn't had common fields added", default: false}
```
Comment (NHJohnson) [inline sub_workflows/kfdrc_lancet_sub_wf.cwl:33]: What input might be a strelka2 vcf?

**Review by NHJohnson — APPROVED** [review]

File: sub_workflows/kfdrc_lancet_sub_wf.cwl (hunk 4 above)
Comment (migbro) [reply sub_workflows/kfdrc_lancet_sub_wf.cwl:33]: Ah, it's kind of copy/paste as it gets fed into the annotation subworkflow. So, this is generic, but I do set the default to fit this case

File: docs/kfdrc_lancet_standalone_wf.md (hunk 1 above)
Comment (migbro) [reply docs/kfdrc_lancet_standalone_wf.md:15]: Ah, no, it's fancy no-period-at-end-of-last-sentence style. Maybe.

**Pull request merged**